### DISCUSSION TITLE
Foreign table query, return as value when only one column

### DISCUSSION THREAD

**Discussion by thiagomendonca-eu** [supabase/supabase#16755] [Questions] [answered]
If I run the following query:
`const example = async () => {
  const { data, error } = await supabase
    .from("tasks")
    .select("*, type:type_id(name)");
};`

The return of the query would be something like:
`task: {
  id: 1,
  name: "Task 1",
  type: {
    name: "Type 1",
  },
}`

My question is, would it be possible to return as follows:
`task: {
  id: 1,
  name: "Task 1",
  type: "Type 1",
}`

**Comment by GaryAustin1** [comment, marked as answer]
https://postgrest.org/en/stable/references/api/resource_embedding.html?highlight=spreading#spread-embedded-resource

And announcement here: https://supabase.com/blog/postgrest-11-prerelease

**Reply by thiagomendonca-eu** [reply]
Very good! Worked, thank you!

`const example = async () => {
  const { data, error } = await supabase
    .from("tasks")
    .select("*, ...type_id(type:name)");
};`

**Reply by thiagomendonca-eu** [reply]
Is it possible to apply a filter to this query, for example: eq("type", "Type 1")?

I am getting the following error when trying this: "column tasks.type does not exist".

**Reply by GaryAustin1** [reply]
You probably need a 2nd aliased reference in the select like `mytypefilter:type_id()` and filter on that.
